### ISSUE TITLE
make messages visible for other apps

### ISSUE THREAD

**Issue by NobodysNightmare** [signalapp/Signal-Android#1659]
I am using TextSecure as my default app for text messages. However, there is a difference between TextSecure and other apps like my phones default or Hangouts:

Using Hangouts or the default app of my phone, I can still access my messages via bluetooth. Third party apps like droidstats or airdroid are able to see the Messages and count them.

Using TextSecure, droidstats does not "see" me sending messages and accessing my messages via Message Access Profile (Bluetooth) or Airdroid only yields messages of the time before I started using TextSecure.

So I believe that TextSecure stores messages differently, though I don't know what the difference is and if it might be a design-decision (e.g. for security reasons). I see that those third-party apps would probably only work with unencrypted messages, but that would be fine for me.

**tl;dr:** Is it possible to make messages sent and received by TextSecure available in the same way as "the default apps" do? I.e. that they are accessible via MAP or third-party apps, even though some of them might be encrypted.

_Disclaimer: I've not developed any app for android, so I have very little knowledge about how things work here and if there is a uniform way to store text messages._

**Comment by JasonKeirstead** [comment]
It would be extremely nice to have an option to store messages in the default SMS database. 

I installed this app to encrypt messages on the wire, not on my device. I don't care if they're encrypted on the device, because I already have full device encryption enabled with secure passcodes and fast wipe.

Not having messages be in the default SMS database breaks so many integrations in Android, that I had to uninstall Signal.
